### PR TITLE
Add static type analysis with PHPStan

### PR DESCRIPTION
**Pull request by oliverklee**
Add Composer scripts for checking the code with PHPStan (while using
a baseline) and for updating the baseline.

When we fix PHPStan-reported issues (or update the PHPStan configuration),
we can update the baseline bit by bit until all issues are fixed.

### CODE REVIEW

**Comment by oliverklee** [comment]
Marking as draft again. I think we should start with level 0 first (so that the baseline file is more manageable), and work our way up through the levels.

I'd also like to add this to the CI build.

**Comment by sabberworm** [comment]
> I'd also like to add this to the CI build.

Feel free to do this. Let the PR CI fail until all code is up to standards.

**Comment by oliverklee** [comment]
> Let the PR CI fail until all code is up to standards.

Even better: With the baseline file, the build will pass with the code as-is, while we still prevent the introduction of new typing issues. Neat, isn't it?

File: phpstan-baseline.neon [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+		-
+			message: "#^Class PHPUnit\\\\Framework\\\\TestCase referenced with incorrect case\\: PHPunit\\\\Framework\\\\TestCase\\.$#"
+			count: 1
+			path: tests/Sabberworm/CSS/OutputFormatTest.php
+
+		-
+			message: "#^Class PHPUnit\\\\Framework\\\\TestCase referenced with incorrect case\\: PHPunit\\\\Framework\\\\TestCase\\.$#"
+			count: 1
+			path: tests/Sabberworm/CSS/ParserTest.php
+
```
Comment (sabberworm) [inline phpstan-baseline.neon:27]: Cool, I thought there’d be more

File: phpstan.neon [hunk 2]
```diff
@@ -0,0 +1,16 @@
+includes:
```
Comment (sabberworm) [inline phpstan.neon:1]: Should we name this `phpstan.neon.dist` as per https://phpstan.org/config-reference#config-file?

**Review by sabberworm — APPROVED** [review]

File: phpstan-baseline.neon (hunk 1 above)
Comment (oliverklee) [reply phpstan-baseline.neon:27]: That's also because we're starting at level 0. (It goes up to level 8 …)

File: phpstan.neon (hunk 2 above)
Comment (oliverklee) [reply phpstan.neon:1]: Thanks! Changed, rebased and repushed.

**Pull request merged**